src/utils/hub-channel.js

```diff
@@ -223,6 +223,18 @@ export default class HubChannel extends EventTarget {
     this.channel.push("update_hub", settings);
   };
 
+  fetchInvite = () => {
+    if (!this._permissions.update_hub) return "unauthorized";
```

Comment:
Looks like we are doing it in all the other calls too but it was a bit surprising that we do this check locally. Is this just an optimization? 

Also extra strange that as a result these functions either return a string or a Promise. I suspect most of the places we are calling these would actually blow up on the string case. Maybe we should start changing these to `Promise.reject("unauthorized")` instead?

Reply:
Yeah, mostly just cargo-culting here. I think I'll just remove the early return, since the server does actually respond appropriately if the user doesn't have permissions.